### DISCUSSION TITLE
Implementing Dynamic Pricing for Jewelry Based on Daily Gold Rates.

### DISCUSSION THREAD

**Discussion by AlkenD** [medusajs/medusa#9421] [Q&A] [answered]
I'm working on building an e-commerce jewelry store where product prices fluctuate. Each product, such as a gold ring, has a base price—let's say $200. In addition to this, I want to factor in the price of gold based on its weight. For example, a ring made from 5g of gold, priced at $100 per gram, would add $500 to the total, making the ring's price $700. Since the price of gold changes daily, I want to be able to update the per-gram price of gold, and have the prices for all relevant products adjust automatically. For instance, if tomorrow the gold price drops to $90 per gram, the same ring would cost $650. How can I implement this dynamic pricing system?

**Comment by radoslaw-sz** [comment, marked as answer]
hi @AlkenD  -  I can let you know that it is possible in Medusa, because my company handled similar use case. We implemented custom pricing for the client who sells products which can be built with the parts and every part influence a price dynamically (Medusa v1).
I cannot give you much details as it was commercial cooperation, but is the high level view:

- we created functionality to set your own attributes which influence a price and put them into metadata of variant
- we created editor where you can pick attributes and create how these attributes influence a price. All information were stored in database in separated table and linked to the variant and attributes.
- we exposed new API for adding such "customizable" products (here it was consulted with the customer, that they can handled it separately on the frontend)
- we created a service which takes attributes and based on the define algorithm, calculates dynamically a price

In your case, you have also additional dynamic attribute which is the gold rate. For sure you would need to have some API which you can call to get the last price of the gold and then you could update database (e.g. on daily basis) which will represent the one of the price attributes.

**Reply by AlkenD** [reply]
Thank you for your insights; they really helped clarify how to implement this feature. Although Is it acceptable to manually update the database by creating custom models and tables? I just didn't want to manually create a table for it but i guess it is unavoidable.

**Reply by radoslaw-sz** [reply]
I think it is unavoidable, unless you will have everything hardcoded in your code :)
